src/Sarif/Baseline/V2/WhereComparer.cs

```diff
@@ -40,21 +40,15 @@ public static int CompareTo(IList<Location> left, Run leftRun, IList<Location> r
 
         public static int CompareTo(Location left, Run leftRun, Location right, Run rightRun)
         {
-            int cmp = 0;
-
             if (left == null && right == null) { return 0; }
             if (left == null) { return -1; }
             if (right == null) { return 1; }
 
-            // Compare by Physical Location, if present
-            cmp = CompareTo(left.PhysicalLocation, leftRun, right.PhysicalLocation, rightRun);
-            if (cmp != 0) { return cmp; }
-
-            // Compare by 'primary' Logical Location, if present
-            cmp = CompareTo(left.LogicalLocation, leftRun, right.LogicalLocation, rightRun);
```

Comment:
>LogicalLocation [](start = 33, length = 15)

The scalar `LogicalLocation` property is actually a convenience property that returns `LogicalLocations[0]`. It exists for backward compatibility because we changes `LogicalLocations` to an array at the last minute, and we didn't want it to be a breaking change.